src/Classical/linear_code.jl

```diff
@@ -425,7 +426,30 @@ function random_information_set(C::AbstractLinearCode)
     return pivot_inds
```

Comment:
I don't entirely see what this second part is doing here. Can you not simply take `permuted_pivot_inds[invperm(original perm)]`?

Also, since we aren't going for speed here, you may as well just do something like `G_perm = C.G[:, perm]` to get your copy.

Reply:
I don't understand what ```permuted_pivot_inds[invperm(original perm)]``` means. 

You might mean to apply invperm to ```permuted_pivot_inds``` through something like ```_col_permutation!``` (RIS_loop function does something like that). The reason it doesn't work here is that ```permuted_pivot_inds``` is a subset of the column indexs. ```invperm``` refers to indexs that do not exist in ```permuted_pivot_inds```

Reply:
Ah. Hmm.... can't you simplify this whole procedure then by doing a perm, rref with col swaps, return col swap perm composed with original perm? You know the info set is in columns 1:k of the rref form, so where do those go under the inverse original perm? For example, make [1, ..., k, 0, ..., 0] then take the nonzero locations of the result. I'm guessing you're trying to do this more efficiently.

Reply:
thanks. This way also has the advantage of actually working